sql/hive/src/test/scala/org/apache/spark/sql/sources/SimpleTextRelation.scala

```diff
@@ -36,7 +38,13 @@ class SimpleTextSource extends TextBasedFileFormat with DataSourceRegister {
       sparkSession: SparkSession,
       options: Map[String, String],
       files: Seq[FileStatus]): Option[StructType] = {
-    Some(DataType.fromJson(options("dataSchema")).asInstanceOf[StructType])
+    val schemaAsString = options("dataSchema")
+    val schema = try {
+      DataType.fromJson(schemaAsString)
+    } catch {
+      case NonFatal(_) => DataType.fromDdl(schemaAsString)
+    }
+    Some(schema.asInstanceOf[StructType])
```

Comment:
`SimpleTextSource` is just a faked source. What is the reason you made these changes?